rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_publisher.hpp

```diff
@@ -75,6 +76,10 @@ class LifecyclePublisher : public LifecyclePublisherInterface,
   publish(std::unique_ptr<MessageT, MessageDeleter> & msg)
   {
     if (!enabled_) {
+      rclcpp::Logger logger = rclcpp::get_logger("LifecyclePublisher");
```

Comment:
Instead of creating a logger every time (with the potential to have typos in the name), I think it would be better to have a private member variable of type `rclcpp::Logger`, initialize it in the constructor, and then just use that for all of the `RCLCPP_WARN` calls.

Reply:
Ok, it worked using a static member variable

Reply:
I don't think we actually want to use a static member variable here; that will get re-initialized any time a new object of this type of class is instantiated.  That may cause havoc for already in-use objects (it is probably not thread-safe, in particular).

Looking at this further, probably the most correct thing to do is to change the `LifeCyclePublisher` constructor to take a type of `rclcpp::node_interfaces::NodeLoggerInterface *` in, then use that in the logging calls.  Unfortunately, this probably also means that we would have to change the `Publisher` constructor to take that as well (since they use the same factory).  If you don't mind giving that a shot, I think that would actually work out a bit better, though it is more code churn.

Reply:
Since each publisher is member of a node (am I right?) I think that automatically receiving the logger of the node that owns it is not a bad thing.

Reply:
> Since each publisher is member of a node (am I right?) I think that automatically receiving the logger of the node that owns it is not a bad thing.

Yep, totally agreed, I think that is the correct thing.  My use of the word "unfortunately" above was merely because it is more churn and work for you :).